DsiplayBuffer error

Hi, just starting to use your extension but when I compile with the latest esphome I get the error:
`AttributeError: module 'esphome.components.display' has no attribute 'DisplayBufferRef'. Did you mean: 'DisplayBuffer'?`
any thoughts?

Hi, sorry for delay, I was away for work. Just tried and get different errors:

```Processing powerdisplay1 (board: esp32dev; framework: arduino; platform: platformio/espressif32@5.4.0)
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
Dependency Graph
|-- AsyncTCP-esphome @ 1.2.2
|-- WiFi @ 2.0.0
|-- FS @ 2.0.0
|-- Update @ 2.0.0
|-- ESPAsyncWebServer-esphome @ 2.1.0
|-- DNSServer @ 2.0.0
|-- ESPmDNS @ 2.0.0
|-- noise-c @ 0.1.4
|-- SPI @ 2.0.0
Compiling .pioenvs/powerdisplay1/src/main.cpp.o
Archiving .pioenvs/powerdisplay1/libe06/libAsyncTCP-esphome.a
Indexing .pioenvs/powerdisplay1/libe06/libAsyncTCP-esphome.a
Archiving .pioenvs/powerdisplay1/lib5b1/libWiFi.a
Indexing .pioenvs/powerdisplay1/lib5b1/libWiFi.a
Archiving .pioenvs/powerdisplay1/lib746/libFS.a
Indexing .pioenvs/powerdisplay1/lib746/libFS.a
Archiving .pioenvs/powerdisplay1/lib4cd/libUpdate.a
Indexing .pioenvs/powerdisplay1/lib4cd/libUpdate.a
src/main.cpp: In function 'void setup()':
src/main.cpp:887:38: error: invalid new-expression of abstract class type 'esphome::st7565::ST7565'
   st7565_st7565 = new st7565::ST7565();
                                      ^
In file included from src/esphome.h:40,
                 from src/main.cpp:3:
src/esphome/components/st7565/st7565.h:12:7: note:   because the following virtual functions are pure within 'esphome::st7565::ST7565':
 class ST7565 : public PollingComponent,
       ^~~~~~
In file included from src/esphome.h:17,
                 from src/main.cpp:3:
src/esphome/components/display/display.h:434:23: note: 	'virtual esphome::display::DisplayType esphome::display::Display::get_display_type()'
   virtual DisplayType get_display_type() = 0;
                       ^~~~~~~~~~~~~~~~
src/main.cpp:898:56: error: cannot declare parameter 'it' to be of abstract type 'esphome::display::DisplayBuffer'
   st7565_st7565->set_writer([=](display::DisplayBuffer it) -> void {
                                 ~~~~~~~~~~~~~~~~~~~~~~~^~
In file included from src/esphome.h:18,
                 from src/main.cpp:3:
src/esphome/components/display/display_buffer.h:15:7: note:   because the following virtual functions are pure within 'esphome::display::DisplayBuffer':
 class DisplayBuffer : public Display {
       ^~~~~~~~~~~~~
In file included from src/esphome.h:17,
                 from src/main.cpp:3:
src/esphome/components/display/display.h:434:23: note: 	'virtual esphome::display::DisplayType esphome::display::Display::get_display_type()'
   virtual DisplayType get_display_type() = 0;
                       ^~~~~~~~~~~~~~~~
In file included from src/esphome.h:18,
                 from src/main.cpp:3:
src/esphome/components/display/display_buffer.h:25:15: note: 	'virtual int esphome::display::DisplayBuffer::get_height_internal()'
   virtual int get_height_internal() = 0;
               ^~~~~~~~~~~~~~~~~~~
src/esphome/components/display/display_buffer.h:26:15: note: 	'virtual int esphome::display::DisplayBuffer::get_width_internal()'
   virtual int get_width_internal() = 0;
               ^~~~~~~~~~~~~~~~~~
src/esphome/components/display/display_buffer.h:29:16: note: 	'virtual void esphome::display::DisplayBuffer::draw_absolute_pixel_internal(int, int, esphome::Color)'
   virtual void draw_absolute_pixel_internal(int x, int y, Color color) = 0;
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
powerdisplay1.yaml:63:4: error: no matching function for call to 'esphome::st7565::ST7565::set_writer(setup()::<lambda(esphome::display::DisplayBuffer)>)'
In file included from src/esphome.h:17,
                 from src/main.cpp:3:
src/esphome/components/display/display.h:411:8: note: candidate: 'void esphome::display::Display::set_writer(esphome::display::display_writer_t&&)'
   void set_writer(display_writer_t &&writer);
        ^~~~~~~~~~
src/esphome/components/display/display.h:411:8: note:   no known conversion for argument 1 from 'setup()::<lambda(esphome::display::DisplayBuffer)>' to 'esphome::display::display_writer_t&&' {aka 'std::function<void(esphome::display::Display&)>&&'}
Compiling .pioenvs/powerdisplay1/libf52/ESPAsyncWebServer-esphome/WebAuthentication.cpp.o
Compiling .pioenvs/powerdisplay1/libf52/ESPAsyncWebServer-esphome/WebHandlers.cpp.o
Compiling .pioenvs/powerdisplay1/libf52/ESPAsyncWebServer-esphome/WebRequest.cpp.o
Compiling .pioenvs/powerdisplay1/libf52/ESPAsyncWebServer-esphome/WebResponses.cpp.o
Compiling .pioenvs/powerdisplay1/libf52/ESPAsyncWebServer-esphome/WebServer.cpp.o
Compiling .pioenvs/powerdisplay1/libf24/DNSServer/DNSServer.cpp.o
Compiling .pioenvs/powerdisplay1/lib026/ESPmDNS/ESPmDNS.cpp.o
Compiling .pioenvs/powerdisplay1/libbdf/libsodium/crypto_aead/chacha20poly1305/sodium/aead_chacha20poly1305.c.o
Compiling .pioenvs/powerdisplay1/libbdf/libsodium/crypto_core/ed25519/core_ed25519.c.o
Compiling .pioenvs/powerdisplay1/libbdf/libsodium/crypto_core/ed25519/core_ristretto255.c.o
Compiling .pioenvs/powerdisplay1/libbdf/libsodium/crypto_core/ed25519/ref10/ed25519_ref10.c.o
*** [.pioenvs/powerdisplay1/src/main.cpp.o] Error 1
```

```
INFO ESPHome 2023.11.6
INFO Reading configuration /config/esphome/st7565.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing st7565 (board: esp01_1m; framework: arduino; platform: platformio/espressif8266@3.2.0)
--------------------------------------------------------------------------------
HARDWARE: ESP8266 80MHz, 80KB RAM, 1MB Flash
Dependency Graph
|-- ESPAsyncTCP-esphome @ 2.0.0
|-- ESPAsyncWebServer-esphome @ 3.1.0
|-- DNSServer @ 1.1.1
|-- ESP8266WiFi @ 1.0
|-- ESP8266mDNS @ 1.2
|-- noise-c @ 0.1.4
|-- SPI @ 1.0
Compiling .pioenvs/st7565/src/main.cpp.o
Compiling .pioenvs/st7565/libbda/ESPAsyncTCP-esphome/SyncClient.cpp.o
Compiling .pioenvs/st7565/libbda/ESPAsyncTCP-esphome/tcp_axtls.c.o
Archiving .pioenvs/st7565/libbda/libESPAsyncTCP-esphome.a
Compiling .pioenvs/st7565/libaf0/Hash/Hash.cpp.o
In file included from src/main.cpp:34:
src/st7565.h:12:7: error: redefinition of 'class esphome::st7565::ST7565'
   12 | class ST7565 : public PollingComponent,
      |       ^~~~~~
In file included from src/esphome.h:42,
                 from src/main.cpp:3:
src/esphome/components/st7565/st7565.h:12:7: note: previous definition of 'class esphome::st7565::ST7565'
   12 | class ST7565 : public PollingComponent,
      |       ^~~~~~
src/main.cpp: In function 'void setup()':
src/main.cpp:897:38: error: invalid new-expression of abstract class type 'esphome::st7565::ST7565'
  897 |   st7565_st7565 = new st7565::ST7565();
      |                                      ^
In file included from src/esphome.h:42,
                 from src/main.cpp:3:
src/esphome/components/st7565/st7565.h:12:7: note:   because the following virtual functions are pure within 'esphome::st7565::ST7565':
   12 | class ST7565 : public PollingComponent,
      |       ^~~~~~
In file included from src/esphome.h:17,
                 from src/main.cpp:3:
src/esphome/components/display/display.h:434:23: note:     'virtual esphome::display::DisplayType esphome::display::Display::get_display_type()'
  434 |   virtual DisplayType get_display_type() = 0;
      |                       ^~~~~~~~~~~~~~~~
src/main.cpp:907:56: error: cannot declare parameter 'it' to be of abstract type 'esphome::display::DisplayBuffer'
  907 |   st7565_st7565->set_writer([=](display::DisplayBuffer it) -> void {
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~^~
In file included from src/esphome.h:18,
                 from src/main.cpp:3:
src/esphome/components/display/display_buffer.h:15:7: note:   because the following virtual functions are pure within 'esphome::display::DisplayBuffer':
   15 | class DisplayBuffer : public Display {
      |       ^~~~~~~~~~~~~
In file included from src/esphome.h:17,
                 from src/main.cpp:3:
src/esphome/components/display/display.h:434:23: note:     'virtual esphome::display::DisplayType esphome::display::Display::get_display_type()'
  434 |   virtual DisplayType get_display_type() = 0;
      |                       ^~~~~~~~~~~~~~~~
In file included from src/esphome.h:18,
                 from src/main.cpp:3:
src/esphome/components/display/display_buffer.h:25:15: note:     'virtual int esphome::display::DisplayBuffer::get_height_internal()'
   25 |   virtual int get_height_internal() = 0;
      |               ^~~~~~~~~~~~~~~~~~~
src/esphome/components/display/display_buffer.h:26:15: note:     'virtual int esphome::display::DisplayBuffer::get_width_internal()'
   26 |   virtual int get_width_internal() = 0;
      |               ^~~~~~~~~~~~~~~~~~
src/esphome/components/display/display_buffer.h:29:16: note:     'virtual void esphome::display::DisplayBuffer::draw_absolute_pixel_internal(int, int, esphome::Color)'
   29 |   virtual void draw_absolute_pixel_internal(int x, int y, Color color) = 0;
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/config/esphome/st7565.yaml:61:4: error: cannot convert 'setup()::<lambda(esphome::display::DisplayBuffer)>' to 'esphome::display::display_writer_t&&' {aka 'std::function<void(esphome::display::Display&)>&&'}
In file included from src/esphome.h:17,
                 from src/main.cpp:3:
src/esphome/components/display/display.h:411:38: note:   initializing argument 1 of 'void esphome::display::Display::set_writer(esphome::display::display_writer_t&&)'
  411 |   void set_writer(display_writer_t &&writer);
      |                   ~~~~~~~~~~~~~~~~~~~^~~~~~
Archiving .pioenvs/st7565/libaf0/libHash.a
Compiling .pioenvs/st7565/lib67b/ESP8266WiFi/BearSSLHelpers.cpp.o
*** [.pioenvs/st7565/src/main.cpp.o] Error 1
========================== [FAILED] Took 6.35 seconds ==========================
```

--- a/components/st7565/display.py
+++ b/components/st7565/display.py
@@ -44,7 +44,7 @@ async def to_code(config):
 
     if CONF_LAMBDA in config:
         lambda_ = await cg.process_lambda(
-            config[CONF_LAMBDA], [(display.DisplayBufferRef, "it")], return_type=cg.void
+            config[CONF_LAMBDA], [(display.DisplayBuffer, "it")], return_type=cg.void
         )
         cg.add(var.set_writer(lambda_))
 

